api/core/key.go

```diff
@@ -254,12 +255,20 @@ func (v *Value) AsInterface() interface{} {
 }
 
 func (v *Value) Emit() string {
-	if v.Type() == STRING {
+	switch v.Type() {
+	case BOOL:
+		return strconv.FormatBool(v.AsBool())
+	case INT32, INT64:
```

Comment:
Not sure I like pairing `INT32` and `AsInt64()` as it relies on an implementation detail about how this value type is implemented. Same with `UINT32` and `AsUint64()`. I suppose it's fine anyway, the tests seem to still pass, because all this is basically about casting.

Reply:
Do you mean the implementation of `AsInt64`? Would you prefer if I a split this into different `case` and change it to `strconv.FormatInt(int64(v.AsInt32()), 10)` and `strconv.FormatUint(uint64(v.AsUint32()), 10)`, respectively?

Reply:
It may sound like a PITA, but yes, please.

Reply:
done :)